Pinta.Effects/Effects/TwistEffect.cs

```diff
@@ -64,79 +64,105 @@ protected override void Render (ImageSurface source, ImageSurface destination, R
 	}
 
 	private static ColorBgra GetFinalPixelColor (
-		TwistSettings settings,
+		in TwistSettings settings,
 		ImageSurface source,
 		ReadOnlySpan<ColorBgra> sourceData,
 		PixelOffset pixel)
 	{
-		PointF offsetFromCenter = new (
-			X: pixel.coordinates.X - (settings.HalfWidth + settings.RenderBounds.Left),
-			Y: pixel.coordinates.Y - (settings.HalfHeight + settings.RenderBounds.Top));
+		PointD fromCenter = new (
+			X: pixel.coordinates.X - settings.Center.X,
+			Y: pixel.coordinates.Y - settings.Center.Y);
 
-		if (offsetFromCenter.MagnitudeSquaredF () > (settings.Maxrad + 1) * (settings.Maxrad + 1))
+		if (fromCenter.MagnitudeSquared () > settings.DistanceThresholdSquared)
 			return sourceData[pixel.memoryOffset];
 
 		int antialiasSamples = settings.AntialiasPoints.Length;
-
 		ColorBgra.Blender aggregate = new ();
 
-		for (int p = 0; p < antialiasSamples; ++p) {
+		for (int i = 0; i < antialiasSamples; ++i) {
+			PointD offset = settings.AntialiasPoints[i];
+			PointD location = fromCenter + offset;
+			aggregate += GetSampleColor (
+				settings,
+				source,
+				sourceData,
+				sourceData[pixel.memoryOffset],
+				location);
+		}
 
-			PointF samplingOffset = settings.AntialiasPoints[p];
-			PointF samplingLocation = offsetFromCenter + samplingOffset;
+		return aggregate.Blend ();
+	}
 
-			double radialDistance = samplingLocation.Magnitude ();
-			double originalTheta = Math.Atan2 (samplingLocation.Y, samplingLocation.X);
-			double radialFactor = 1 - radialDistance / settings.Maxrad;
-			double twistAmount = (radialFactor < 0) ? 0 : (radialFactor * radialFactor * radialFactor);
-			double twistedTheta = originalTheta + (twistAmount * settings.Twist / 100);
+	private static ColorBgra GetSampleColor (
+		in TwistSettings settings,
+		ImageSurface source,
+		ReadOnlySpan<ColorBgra> sourceData,
+		ColorBgra original,
+		PointD location)
+	{
+		double radialDistance = location.Magnitude ();
 
-			PointI samplePosition = new (
-				X: (int) (settings.HalfWidth + settings.RenderBounds.Left + (float) (radialDistance * Math.Cos (twistedTheta))),
-				Y: (int) (settings.HalfHeight + settings.RenderBounds.Top + (float) (radialDistance * Math.Sin (twistedTheta)))
-			);
+		// Prevent division by zero at center point (which might prevent the rest of the row from rendering)
+		if (radialDistance == 0)
```

Comment:
Is this check still needed? I don't see a division by `radialDistance` around anymore